cmd/rebalance-admin.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type rebalPoolProgress struct {
+	NumObjects  uint64        `json:"objects"`
+	NumVersions uint64        `json:"versions"`
+	Bytes       uint64        `json:"bytes"`
+	Bucket      string        `json:"bucket"`
+	Object      string        `json:"object"`
+	Elapsed     time.Duration `json:"elapsed"`
+	ETA         time.Duration `json:"eta"`
+}
+
+type rebalancePoolStatus struct {
+	ID       int               `json:"id"`                 // Pool index (zero-based)
+	Status   string            `json:"status"`             // Active if rebalance is running, empty otherwise
+	Used     float64           `json:"used"`               // Percentage used space
+	Progress rebalPoolProgress `json:"progress,omitempty"` // is empty when rebalance is not running
+}
+
+// rebalanceAdminStatus holds rebalance status related information exported to mc, console, etc.
+type rebalanceAdminStatus struct {
+	ID        uuid.UUID             // identifies the ongoing rebalance operation by a uuid
+	Pools     []rebalancePoolStatus `json:"pools"` // contains all pools, including inactive
+	StoppedAt time.Time             `json:"stoppedAt,omitempty"`
+}
+
+func rebalanceStatus(ctx context.Context, z *erasureServerPools) (r rebalanceAdminStatus, err error) {
+	// Load latest rebalance status
+	meta := &rebalanceMeta{}
+	err = meta.load(ctx, z.serverPools[0])
+	if err != nil {
+		return r, err
+	}
+
+	// Compute disk usage percentage
+	si, _ := z.StorageInfo(ctx)
+	diskStats := make([]struct {
+		AvailableSpace uint64
+		TotalSpace     uint64
+	}, len(z.serverPools))
+	for _, disk := range si.Disks {
+		diskStats[disk.PoolIndex].AvailableSpace += disk.AvailableSpace
+		diskStats[disk.PoolIndex].TotalSpace += disk.TotalSpace
+	}
+
+	stopTime := meta.StoppedAt
+	r = rebalanceAdminStatus{
+		ID:        uuid.UUID(meta.ID),
```

Comment:
```suggestion
		ID:        uuid.UUID(meta.ID),
```

You can use something called as `shortuuid.New()` which is nicer than uuid.UUID()  its shorter and nicer.